kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActionsIterator.java

```diff
@@ -174,12 +175,17 @@ private CloseableIterator<ActionWrapper> getNextActionsIter() {
 
                 return combine(dataIter, false /* isFromCheckpoint */, fileVersion);
             } else if (nextFile.getPath().endsWith(".parquet")) {
-                final long fileVersion = FileNames.checkpointVersion(nextFile.getPath());
+                final long fileVersion = checkpointVersion(nextFile.getPath());
+
+                // Try to retrieve the remaining checkpoint files (if there are any) and issue
+                // read request for all in one go, so that the {@link ParquetHandler} can do
+                // optimizations like reading multiple files in parallel.
+                CloseableIterator<FileStatus> checkpointFilesIter =
+                        retrieveRemainingCheckpointFiles(nextFile, fileVersion);
```

Comment:
will the checkpoint files for the same version will always be contiguous in the list?

Reply:
That is the expectation of input file list.